app/helpers/application_helper.rb

```diff
@@ -58,16 +58,19 @@ def mobile_needs_verification?
   # will be shown but the dialog is not shown automatically.
 
   def log_in_with_facebook?
+    return false if %w[all facebook].include? ENV["DISABLE_LOG_INS"]
     return true unless params[:log_in_with]
     return %w[any facebook].include? params[:log_in_with]
   end
 
   def log_in_with_twitter?
+    return false if %w[all twitter].include? ENV["DISABLE_LOG_INS"]
```

Comment:
Why disable Twitter if `ENV["DISABLE_LOG_INS"] == "facebook"`?

Reply:
it's not, this line will disable twitter if `ENV["DISABLE_LOG_INS"] == "twitter"`

Reply:
OK you can tell I haven't had enough sleep recently ... :man_facepalming: